### ISSUE TITLE
Allowing more than one currency for an account

### ISSUE THREAD

**Issue by dfrochtm** [moneymanagerex/moneymanagerex#4123]
# Description of the feature

Allow to use more than one currency per account.

# Use cases

Having created the account with its main currency, the idea is to be able to add transactions with another currency.

Rate change would apply for each transaction from the foreign currency to the default.

**Comment by whalley** [comment]
Not sure of the exact use case for this can you elaborate? What real world accounts actually work like this?

If I make a foreign currency transaction on any of my accounts it gets reflected in my account as a transaction in the base currency with the bank doing the rate change (with perhaps a fee). If you wanted to record the actual original foreign currency value you can use the notes field.

**Comment by Dandy64** [comment]
I have created an account with the main currency and called it **Bank_Main**.
In addition, I have 2 "virtual" accounts - **Bank_Euro** and **Bank_Dollar**. Actually all the above is the same bank account, but this separation allows me to be flexible in my transactions.
For example, if I need I can transfer 100$ from my **Bank_Dollar** to the **Bank_Main**. And I can see this transaction in the report.
If I buy something and _pay_ 100$, it looks like withdrawal from my **Bank_Dollar**.

**Comment by dfrochtm** [comment]
@whalley I agree that no real world accounts work like this. But as explained by @Dandy64, today the only solution is to create virtual accounts and then transfer transactions.

At that time only 1 software I have used allow to create transaction with another currency and applies the rate change to the default currency.

e.g : you travel worldwide and spend weeks or months abroad. It becomes more confortable to enter transactions using the local currency that will be automatically converted to your main currency. I agree that bank fees can't be anticipated.

There are come (few) countries where local currencies rate don't change compared to EUR. (See XFP for example).

**Comment by whalley** [comment]
> e.g : you travel worldwide and spend weeks or months abroad. It becomes more confortable to enter transactions using the local currency that will be automatically converted to your main currency. I agree that bank fees can't be anticipated.

The application will of course not know the currency conversion rate that will be applied by the bank so could only use the latest rates known to the application. I guess one option might be to provide an advanced option to allow a currency to be specified when entering the transaction and then do the conversion to account currency.

However given that we have 'workarounds' as @Dandy64 has mentioned, and this being a feature that would be very rarely used I don't see the addition of anything to support this happening in the near term. I could of course be wrong and somebody might pick it up.

**Comment by Dandy64** [comment]
I even wouldn't call it workarounds :-) The fact is my bank works exactly in such a manner.
When I open a new account it will be open in local currency. If I wanna add US$ into my account, the bank opens a special foreign currency account. But actually, this is the same account with the same number!

So let's say I have 100 units in my local currency and 100 US$ and **US$1=3 units in local currency** at the moment.
Before I leave my country for the US, I can give instructions to my bank - how I want to pay my expenses when I am abroad. 
I can request to use my dollar account, so if I pay let's say $10 they will be taken from my currency account without any conversions. **US$100 - US$10 = US$90**
But by default, without my instructions, the bank takes money from the account with local currency. So $10 will be converted to the local currency (yes, I dunno the exact rate) and it will be taken from let's say local part of the account. 
**US$10 * 3.2 (unknown bank's conversion rate now) = 32 units in local currency**.
So when I come back I will find that I paid the sum equivalent to the conversion rate for the date of my purchasing.
**100 units in local currency - 32 = 68 units.**
But in reality, I do not care what the conversion rate was since I have nothing to do with this information and I don't need to enter it into Money Manager EX.

**Comment by tactilis** [comment]
@Dandy64
I'm interested. Which bank is this and what's the name of the account please?

**Comment by Dandy64** [comment]
Israel. Bank Hapoalim
https://www.bankhapoalim.com/en

**Comment by Dandy64** [comment]
Israeli banks offer two common types of accounts to new residents and foreigners: current accounts and savings accounts.

Current accounts are called over v’shav in Hebrew and are used for bill pay, money transfers and everyday transactions. Savings accounts can have fixed or variable rates. Depending on your account and your interest rates, you may have restricted access to the currency in a savings account.

To open a bank account as new immigrant or foreign resident, you must bring your resident identification card (teudat zehut), passport or visa. Banks require a small fee to open the account and an initial deposit (100 shekels is enough). To make the process quicker and establish better credit with your bank, bring recent statements from other bank accounts.

You will typically receive a debit card when you open an account which can be used at ATM machines (caspomat) throughout Israel. Many banks will charge a monthly service fee instead of charging every transaction made with a debit card. As a result, Israelis typically use debit cards for most purchases and payments.

Banks like Leumi or Discount Bank of Israel will give you a credit card if you have a significant initial deposit (more than 2,500 shekels). These cards are for use in Israel only. You must apply for an international credit card and pay additional fees if you use it abroad.

Banks in Israel do not generally give you cheques upon opening your account. You may apply for cheques after you have held your account for three months. Depending on the bank and which services you have, the bank may charge you for chequebooks.

Foreign currency accounts
If you want to retain some of your money in its current currency but have it in Israel, you can open a foreign currency account. Most Israeli banks will allow you to open savings and current accounts in a foreign currency. You are permitted to purchase shekels using your foreign currency at the current exchange rate. Banks charge a small service fee for every transaction and for purchasing shekels with your foreign currency.

To save receive higher interest rates on your savings, you can lock your account. Locking the account for a set period of time means that your account will earn a rate of interest for a fixed period of time. You will not be able to access the money in your account during this time (unless you want to pay large fees).

**Comment by Dandy64** [comment]
Israeli banks offer two common types of accounts to new residents and foreigners: current accounts and savings accounts.
Current accounts are called over v’shav in Hebrew and are used for bill pay, money transfers and everyday transactions. Savings accounts can have fixed or variable rates. Depending on your account and your interest rates, you may have restricted access to the currency in a savings account.
Foreign currency accounts
If you want to retain some of your money in its current currency but have it in Israel, you can open a foreign currency account. Most Israeli banks will allow you to open savings and current accounts in a foreign currency. You are permitted to purchase shekels using your foreign currency at the current exchange rate. Banks charge a small service fee for every transaction and for purchasing shekels with your foreign currency.
To save receive higher interest rates on your savings, you can lock your account. Locking the account for a set period of time means that your account will earn a rate of interest for a fixed period of time. You will not be able to access the money in your account during this time (unless you want to pay large fees).

**Comment by Superpila** [comment]
Another possible case is with accounts on cryptocurrencies exchange. Each account handles many currencies that can be purchased, sold, deposited, withdrawn and swapped while remaining within your account.

**Comment by tactilis** [comment]
> Another possible case is with accounts on cryptocurrencies exchange. Each account handles many currencies that can be purchased, sold, deposited, withdrawn and swapped while remaining within your account.

Maybe - but the same situation as described previously would apply: MMEX cannot know what exchange rate will be applied to the foreign currency transaction, so you are always going to have to adjust the transaction manually.

**Comment by Superpila** [comment]
It's true.
But there is already advanced transfer, adding an optional field "fee" the problem would be easily solved, I suppose.
